### ISSUE TITLE
unable to run pipeline 2-environments after 3-networks-hub-and-spoke has run

### ISSUE THREAD

**Issue by TjenWellens** [terraform-google-modules/terraform-example-foundation#974]
### TL;DR

environments_service_account is blocked from `terraform plan` by VPC service controls.

### Expected behavior

`terraform plan` works in pipelines

### Observed behavior

`terraform plan` crashes with

> | Error: Error when reading or editing Service Account "projects/prj-d-shared-restricted-[redacted]/serviceAccounts/project-service-account@prj-d-shared-restricted-[redacted].iam.gserviceaccount.com": googleapi: Error 403: Request is prohibited by organization's policy. vpcServiceControlsUniqueIdentifier: [redacted]
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.PreconditionFailure",
│     "violations": [
│       {
│         "description": "[redacted]",
│         "type": "VPC_SERVICE_CONTROLS"
│       }
│     ]
│   },
│   {
│     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
│     "domain": "googleapis.com",
│     "metadata": {
│       "consumer": "projects/[redacted]",
│       "service": "iam.googleapis.com",
│       "uid": "[redacted]"
│     },
│     "reason": "SECURITY_POLICY_VIOLATED"
│   }
│ ]
│ , forbidden
│ 
│   with module.env.module.restricted_shared_vpc_host_project.module.project-factory.

### Terraform Configuration

```hcl
I assume it's reproduced by running 

0-bootstrap
1-org
2-environments
3-networks-hub-and-spoke

and then running the Cloud Build pipeline for
2-environments

assuming you did not manually add the service account of step-2 to 3-networks-hub-and-spoke's `common.auto.tfvars`
```

### Terraform Version

```sh
Terraform v1.3.9
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v4.68.0
+ provider registry.terraform.io/hashicorp/google-beta v4.68.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/time v0.9.1

(but it's also on the Cloud Build pipeline)
```

### Additional information

PR incoming
